### PR TITLE
Usa case tag

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/34198

## Description
* Added a validation that the Use Case tag in pack_metadata file is only used when the pack contains at least one PB, Incident Type or Layout.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 1]
```diff
@@ -181,7 +181,8 @@ def validate_pack_meta_file(self):
             self._is_approved_tags(),
             self._is_price_changed(),
             self._is_approved_tags(),
-            self._is_valid_support_type()
+            self._is_valid_support_type(),
+            self.is_right_usage_of_usecase_tag()
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:185]: [PEP8](https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas)
```suggestion
            self.is_right_usage_of_usecase_tag(),
```

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 2]
```diff
@@ -350,6 +351,33 @@ def _is_approved_tags(self) -> bool:
...
+
+            if "Use Case" in pack_meta_file_content['tags']:
+                playbooks_path = self.pack_path + "Playbooks"
+                incidents_path = self.pack_path + "IncidentTypes"
+                layouts_path = self.pack_path + "Layouts"
+
+                answers = [
+                    os.path.exists(playbooks_path) and len(os.listdir(playbooks_path)) != 0,
+                    os.path.exists(incidents_path) and len(os.listdir(incidents_path)) != 0,
+                    os.path.exists(layouts_path) and len(os.listdir(layouts_path)) != 0
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:371]: PEP8
```suggestion
                    os.path.exists(layouts_path) and len(os.listdir(layouts_path)) != 0,
```

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 3]
```diff
@@ -350,6 +351,33 @@ def _is_approved_tags(self) -> bool:
...
+        Return:
+             bool: True if the Pack contains at least one PB, Incident Type or Layout, otherwise False
+        """
+        try:
+            pack_meta_file_content = json.loads(self._read_file_content(self.pack_meta_file))
+
+            if "Use Case" in pack_meta_file_content['tags']:
+                playbooks_path = self.pack_path + "Playbooks"
+                incidents_path = self.pack_path + "IncidentTypes"
+                layouts_path = self.pack_path + "Layouts"
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:366]: use `os.path.join`

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 4]
```diff
@@ -259,6 +259,36 @@ def test_is_approved_tags(self, repo, tags, is_valid):
...
+            playbook = pack.create_playbook(name="PlaybookName")
+            pack.playbooks.append(playbook)
+        elif pack_content == "incident":
+            incident_type = pack.create_incident_type(name="IncidentTypeName")
+            pack.incident_types.append(incident_type)
+        elif pack_content == "layout":
+            layout = pack.create_layout(name="Layout")
+            pack.layouts.append(layout)
+
+        self.validator.pack_path = pack.path + "/"
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:289]: why do you need to append the slash?

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 5]
```diff
@@ -259,6 +259,36 @@ def test_is_approved_tags(self, repo, tags, is_valid):
...
+        ("layout", ["Use Case"], True),
+        ("playbook", [], True),
+    ])
+    def test_is_right_usage_of_usecase_tag(self, repo, pack_content, tags, is_valid):
+        pack_name = 'PackName'
+        pack = repo.create_pack(pack_name)
+        pack.pack_metadata.write_json({
+            PACK_METADATA_USE_CASES: [],
+            PACK_METADATA_SUPPORT: XSOAR_SUPPORT,
+            PACK_METADATA_TAGS: tags
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:276]:
```suggestion
            PACK_METADATA_TAGS: tags,
```

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 6]
```diff
@@ -259,6 +259,36 @@ def test_is_approved_tags(self, repo, tags, is_valid):
...
+        })
+
+        if pack_content == "playbook":
+            playbook = pack.create_playbook(name="PlaybookName")
+            pack.playbooks.append(playbook)
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:281]: redundant, the `create_playbook` function already append it (same for the others).

Comment (merit-maita) [reply demisto_sdk/commands/common/tests/pack_unique_files_test.py:281]: i did it first without and the test failed, while debugging it showed that it doesn't have any playbooks

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py (hunk 4 above)
Comment (merit-maita) [reply demisto_sdk/commands/common/tests/pack_unique_files_test.py:289]: pack.path doesnt contain /, without it the test fails

Comment (yaakovi) [reply demisto_sdk/commands/common/tests/pack_unique_files_test.py:289]: now that the function uses `os.path.join` it should solve that error

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py (hunk 6 above)
Comment (yaakovi) [reply demisto_sdk/commands/common/tests/pack_unique_files_test.py:281]: this sounds weird, can you please double-check (add a screenshot)

**Review by yaakovi — COMMENTED** [review]
see my replies in the unresolved comments

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py (hunk 6 above)
Comment (merit-maita) [reply demisto_sdk/commands/common/tests/pack_unique_files_test.py]: i did double check and it worked, weird, just fixed it.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**